internal/clients/kibana/alerting.go

```diff
@@ -124,18 +130,26 @@ func UpdateAlertingRule(ctx context.Context, apiClient *clients.ApiClient, rule
 		Tags:     rule.Tags,
 		Throttle: *alerting.NewNullableString(rule.Throttle),
 	}
+
 	req := client.UpdateRule(ctxWithAuth, rule.RuleID, rule.SpaceID).KbnXsrf("true").UpdateRuleRequest(reqModel)
+
 	ruleRes, res, err := req.Execute()
 	if err != nil && res == nil {
 		return nil, diag.FromErr(err)
 	}
-	rule.RuleID = ruleRes.Id
+
+	if ruleRes != nil {
+		rule.RuleID = ruleRes.Id
+	}
+
 	defer res.Body.Close()
+
 	if diags := utils.CheckHttpError(res, "Unable to update alerting rule"); diags.HasError() {
 		return nil, diags
 	}
 
 	shouldBeEnabled := rule.Enabled != nil && *rule.Enabled
+
 	if shouldBeEnabled && !ruleRes.Enabled {
```

Comment:
We're assuming here that anything where `ruleRes == nil` will be caught by `utils.CheckHttpError` above? I wonder if it's worth having an explicit check as well?

Reply:
> We're assuming here that anything where ruleRes == nil will be caught by utils.CheckHttpError above?

Reply:
I've added logic to explicitly handle an empty response with HTTP 200.

Reply:
Sorry - the new logic just returns an error in case of an empty response for Create/Update call, HTTP status code doesn't make any diff in this case. 
Still in reality, `CheckHttpError` should handle the cases when something goes wrong based on the HTTP status code. The new logic just makes sure that crash won't happen.